hydrator-common/src/main/java/io/cdap/plugin/common/Asset.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.common;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a dataset with a FQN which is a fully-qualified unique identifier for a dataset
+ * and the location of the dataset.
+ */
+public class Asset {
+    private final String fqn;
+    private final Map<String, String> properties;
```

Comment:
Here I think we can expose the location information as a field, we can just change that to a Map for `Endpoint` in `LineageRecorder`, if you have a Map here and have the plugin to construct this Map, you won't guarantee each plugin will use the correct key for DC to use

Reply:
Sounds right